keel-core/src/main/kotlin/com/netflix/spinnaker/keel/enforcers/EnvironmentExclusionEnforcer.kt

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.keel.enforcers
+
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.verification.VerificationContext
+import org.springframework.stereotype.Component
+
+/**
+ * Exception thrown when it's not safe to take action against the environment because
+ * something is is acting on it
+ */
+class EnvironmentCurrentlyBeingActedOn(message: String) : Exception(message) { }
+
+@Component
+class EnvironmentExclusionEnforcer {
```

Comment:
Please consider adding a class doc. The description from your PR would be great here.